src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CostServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.companieshouse.api.accounts.service.impl;
+
+import java.util.List;
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.accounts.PayableResource;
+import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.exception.ServiceException;
+import uk.gov.companieshouse.api.accounts.model.rest.Cost;
+import uk.gov.companieshouse.api.accounts.model.rest.Costs;
+import uk.gov.companieshouse.api.accounts.service.CostService;
+import uk.gov.companieshouse.api.accounts.service.TransactionService;
+import uk.gov.companieshouse.api.accounts.utility.YamlResourceMapper;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+
+@Service
+public class CostServiceImpl implements CostService {
+
+    @Autowired
+    private TransactionService transactionService;
+
+    @Autowired
+    private YamlResourceMapper yamlResourceMapper;
+
+    private static final String COSTS_YAML_FILE = "/costs/costs.yaml";
+
+    private Costs costs;
+
+    @Override
+    public Cost[] getCosts(Transaction transaction) throws DataException {
+
+        try {
+            List<PayableResource> payableResources = transactionService.getPayableResources(transaction);
+
+            if (payableResources.isEmpty()) {
+                return new Cost[0];
+            }
+
+            if (costs == null) {
+                costs =
```

Comment:
This is config and should be read in at the start as opposed to on its first use, if its fails then we know theres an issue instantly.